src/unitxt/metric.py

```diff
@@ -98,11 +99,16 @@ def process(self, predictions: List[str], references: Iterable, split_name: str
 from .schema import UNITXT_DATASET_SCHEMA
 
 
+def _from_key_value_pairs(key_value_list: Dict[str, list]) -> Dict[str, str]:
+    return dict([(key, value) for key, value in zip(key_value_list["key"], key_value_list["value"])])
+
+
 class MetricRecipe(SequentialOperatorInitilizer):
     def prepare(self):
         register_all_artifacts()
         self.steps = [
             FromPredictionsAndOriginalData(),
+            Apply("additional_inputs", function=_from_key_value_pairs, to_field="additional_inputs"),
```

Comment:
again, maybe document a bit what is going on. AFAIU this operator processes the "additional_inputs" field and converts it from the two keyed-dictionary to a dict where the keys are.. the input and output fields?

Reply:
Added documentation.

Regarding the second questions, . It is the field from which the value the Apply function receives is called.